### PR TITLE
Make Lavender Essence cancel out miasma

### PR DESCRIPTION
**Pull request by TheTyrant25**
[CHEMISTRY]
## About the PR
Makes lavender essence and miasma reagents destroy each other.
May or may not work. havent coded in a while. heh

## Why's this needed? 
Seen some people try to flood areas with lavender gas and it didnt work :(

### CODE REVIEW

**Comment by DecidedlyUnethical** [comment]
If you're smoking this, in a rafflesia for example, won't this result in a massive amount of chat spam/reaction messages?

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -3804,6 +3804,18 @@
...
+	miasma_neutralisation
+		name = "miasma neutralisation"
+		id = "miasma_neutralisation"
+		instant = 1
+		required_reagents = list("miasma" = 1, "lavender_essence" = 1)
+		mix_phrase = "The lavender counteracts the miasma."
+		on_reaction(var/datum/reagents/holder, var/created_volume)
+			if (holder)
+				holder.del_reagent("miasma")
+				holder.del_reagent("lavender_essence")
```
Comment (Zonespace27) [inline code/modules/chemistry/Chemistry-Recipes.dm:3817]: this means that 1u of miama and Nu of lavender (or vice versa) will cause both to be fully deleted

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -3804,6 +3804,18 @@
...
+			return
```
Comment (Zonespace27) [inline code/modules/chemistry/Chemistry-Recipes.dm:3818]: uneeded return

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 3]
```diff
@@ -3804,6 +3804,18 @@
 			//This reaction does only happen in carbon-based beings and for only as long as there is less than 15u lungrot in the person
 			return holder.my_atom && iscarbon(holder.my_atom) && (holder.get_reagent_amount("lungrot_bloom") < 15)
 
+	// makes lavender and miasma destroy each other
+	miasma_neutralisation
+		name = "miasma neutralisation"
+		id = "miasma_neutralisation"
+		instant = 1
```
Comment (Zonespace27) [inline code/modules/chemistry/Chemistry-Recipes.dm:3811]:
```suggestion
		instant = TRUE
```

File: code/modules/chemistry/Chemistry-Recipes.dm [hunk 4]
```diff
@@ -3804,6 +3804,18 @@
...
+		required_reagents = list("miasma" = 1, "lavender_essence" = 1)
+		mix_phrase = "The lavender counteracts the miasma."
+		on_reaction(var/datum/reagents/holder, var/created_volume)
```
Comment (Zonespace27) [inline code/modules/chemistry/Chemistry-Recipes.dm:3814]: `var/` isn't needed in proc args

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Chemistry-Recipes.dm (hunk 1 above)
Comment (TheTyrant25) [reply code/modules/chemistry/Chemistry-Recipes.dm:3817]: ah oops

**Comment by TheTyrant25** [comment]
oh i see

**Pull request closed without merging**